src/main/java/redis/clients/jedis/JedisClusterCommand.java

```diff
@@ -100,131 +101,91 @@ public T runWithAnyNode() {
     }
   }
 
-  private boolean shouldBackOff(int attemptsLeft) {
-    int attemptsDone = maxAttempts - attemptsLeft;
-    return attemptsDone >= maxAttempts / 3;
-  }
-
-  private static long getBackoffSleepMillis(int attemptsLeft, Instant deadline) {
-    if (attemptsLeft <= 0) {
-      return 0;
-    }
-
-    long millisLeft = Duration.between(Instant.now(), deadline).toMillis();
-    if (millisLeft < 0) {
-      throw new JedisClusterMaxAttemptsException("Deadline exceeded");
-    }
-
-    return millisLeft / (attemptsLeft * attemptsLeft);
-  }
-
   private T runWithRetries(final int slot) {
     Instant deadline = Instant.now().plus(maxTotalRetriesDuration);
-    Supplier<Jedis> connectionSupplier = () -> connectionHandler.getConnectionFromSlot(slot);
-
-    // If we got one redirection, stick with that and don't try anything else
-    Supplier<Jedis> redirectionSupplier = null;
 
-    for (int currentAttempt = 0; currentAttempt < this.maxAttempts; currentAttempt++) {
+    JedisRedirectionException redirect = null;
+    AtomicInteger consecutiveConnectionFailures = new AtomicInteger();
```

Comment:
```suggestion
    int consecutiveConnectionFailures = 0;
```

Why `AtomicInteger`?

Reply:
For this https://github.com/walles/jedis/pull/2/files#diff-58ec82b0d7b29da6455767ace17ef1c0d26d6e9e539480f1f1cb9b8de86bd516R167

Reply:
Invalidated after update.